bin/cron/hoc_student_name_cleanup

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env ruby
+require_relative 'only_one'
+abort 'Script already running' unless only_one_running?(__FILE__)
+
+# This script deletes the name stored in pegasus.hoc_activity after 3 months.
+#
+# Background:
+# The name field is populated when the user personalizes the student certificate
+# after completing an HoC activity. This allows us to create the personalized
+# certificate when it is requested, e.g. when someone clicks on a link that the
+# student has shared to social media. For privacy reasons, we delete the name after
+# 3 months. If the certificate is requested after that point, we return the default
+# unpersonalized certificate.
+
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require 'cdo/db'
+require 'cdo/properties'
+require 'optparse'
+
+CDO.log.info("Script started at #{Time.now}")
+
+HOC_ACTIVITY = PEGASUS_DB[:hoc_activity]
+START_ID_PROPERTY_KEY = "hoc_activity.name_cleanup.start_id"
+
+# Parse options
+options = {
+  retention_days: 90,
+  retain_after_date: nil, # calculated from retention_days
+  start_id: nil,
+  end_id: nil,
+  batch_limit: 100,
+  sleep_msecs: 100,
+  dry_run: false,
+  log_sql: false,
+}
+OptionParser.new do |opts|
+  opts.banner = <<~BANNER
+    Usage: #{File.basename(__FILE__)} [options]
+
+    This script deletes names stored in the pegasus.hoc_activity table after the
+    configured retention period.
+
+    Options:
+  BANNER
+
+  opts.on('--retention=90',
+    Integer,
+    'Retention period in days. Defaults to 90 days if not specified.'
+  ) do |retention_days|
+    options[:retention_days] = retention_days
+  end
+
+  opts.on('--end-id=1000000',
```

Comment:
Assuming the significance of 1m is just "sufficiently large"?

Reply:
This 1000000 is just an arbitrary number that is displayed in the usage text and not actually used anywhere in the script.  For the other options I used the default value but the default value for end-id is the id of the row + 1 so I just chose an arbitrary number.  Perhaps it would be less confusing if I changed this to 1234567?

Reply:
Ahh yes 1234567 would definitely read as arbitrary! Thanks!